app/controllers/concerns/client_sortable.rb

```diff
@@ -24,4 +28,22 @@ def cookie_filters
 
     cookies[filter_cookie_name] ? HashWithIndifferentAccess.new(JSON.parse(cookies[filter_cookie_name])) : {}
   end
+
+  def vita_partners_for_tagify
+    # This returns the vita partners in a format that the tagify js library can understand
+    vita_partners = @client_sorter.active_filters[:vita_partners]
+    return if vita_partners.blank?
+    vita_partners = JSON.parse(vita_partners)
+    result = vita_partners.map do |id|
+      id = id[:id] if id.instance_of?(Hash)
```

Comment:
Will `id` ever be something other than a hash, and will the final `to_json` call fail if there are any nils in the `result` list (as i believe there could be currently)? I.e. should we call `compact` on result before we call `to_json`? Or handle nils some other way? 

I probably wouldn't have even thought to ask if you hadn't handled the possible `method_missing`, fwiw :)

Reply:
Moving forward, id will always be an integer. The only case where it will be a hash is for folks that have had a cookie defined with the old code. In this case, the array of vita_partners is in the format `{"id": 1234, "name": "Some long partner name", "parent_name": "Some long parent name", "value": 1234}`.

After a while, this check will be able to go away. I noted this in the PR description, but I think I'll add a comment to the code too.